frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/OptionListEditor.test.tsx

```diff
@@ -0,0 +1,171 @@
+    ).toBeInTheDocument();
+  });
+
+  it('should render a error message when api throws an error', async () => {
+    renderOptionListEditor({
+      queries: {
+        getOptionLists: jest.fn().mockRejectedValueOnce(new Error('Error')),
+      },
+    });
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    expect(
+      screen.getByText(textMock('ux_editor.modal_properties_error_message')),
+    ).toBeInTheDocument();
+  });
+
+  it('should render the open Dialog button', async () => {
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    const btnOpen = screen.getByRole('button', {
+      name: textMock('ux_editor.modal_properties_code_list_open_editor'),
+    });
+
+    expect(btnOpen).toBeInTheDocument();
+  });
+
+  it('should open Dialog', async () => {
+    const user = userEvent.setup();
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
+
+  it('should close Dialog', async () => {
+    const user = userEvent.setup();
+    renderOptionListEditor();
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should call handleClose when closing Dialog', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    renderOptionListEditor({
+      previewContextProps: { doReloadPreview },
+    });
+    await waitForElementToBeRemoved(() => {
+      return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
+    });
+
+    await openModal(user);
+
+    await user.click(screen.getByRole('button', { name: 'close modal' })); // Todo: Replace "close modal" with defaultDialogProps.closeButtonTitle when https://github.com/digdir/designsystemet/issues/2195 is fixed
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    expect(doReloadPreview).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call onChange when changing value in a textField', async () => {
+    const user = userEvent.setup();
+    const doReloadPreview = jest.fn();
+    renderOptionListEditor({
+      previewContextProps: { doReloadPreview },
+    });
+    await waitForElementToBeRemoved(() => {
```

Comment:
Instead og repeating this `waitForElementToBeRemoved`, we can use a normal pattern in this codebase which is to have a different render method that does the waiting:

```
const resolveAndWaitForSpinnerToDisappear = async ({
  previewContextProps = {},
  queries = {},
} = {}) => {
  renderOptionListEditor({ previewContextProps, queries });
  await waitForElementToBeRemoved(() => {
    return screen.queryByText(textMock('ux_editor.modal_properties_code_list_spinner_title'));
  });
};
```